doc/source/preparing_releases_and_hotfixes.rst

```diff
@@ -176,6 +176,13 @@ Create a release of the UserGuide
 
 For now, the UserGuide is released at the same time as the core library. To make a release of the UserGuide you should:
 
+#. Update the version of MDA used by the UserGuide to the release version.
+
+#. Re-generate the Syrup test snapshots, and commit those to git and confirm the build passes.
+
+    .. code-block:: bash
+        python -m pytest --snapshot-update
+
```

Comment:
Maybe this needs to be more specific, ie what command to run to commit and push to git? Happy to revise

Reply:
You'd probably want to do this after the release? Since it's not a requirement for ci tests to pass, we could just do a release and just say "remember to run and commit this after"? What actually changes on release?

Reply:
Sorry, i'm guessing progressively confused here. If core MDA changes on "develop", for example by introducing a new reader/writer, this will have to be re-generated. Sorry i feel like i'm going in circles here myself

Reply:
Sorry about the confusion here, this is why I asked about what folks were thinking would happen on release - because I wasn't sure about what had been mentioned in the comments.

The userguide streams against the current develop state of the core library. So my understanding here is that these snapshots will just start failing whenever something in develop fails? In that case, when we get to release we should never have to create new snapshots because we should have already created new ones / fixed the current state of the userguide by the time we got there?

Unless the snapshots contain any release information?

Reply:
So I guess my question is - for now should we just have the docs say "before release make sure these tests pass, if they don't then do this"?

Reply:
Correct, that's a reasonable flow. I'd add that a snapshot update might also be necessary for "regular" changes on MDA core develop branch – because that's what the UserGuide is pulling in, there is limited version pinning.

Reply:
No the snapshots just contain the serialized lines `list[list[str]]` produced by the TableWriter classes. Anytime somebody changes something on develop that impacts these lines, the snapshots will fail. 

Yeah, basically by the time it comes to the release, all the snapshots should be up-to-date with what's on develop.

Reply:
Sorry to be so slow to reply, teething problems 👶

Reply:
Sorry maybe I wasn't clear - my question here was more of an ask, can we reword this as "if it's failing please fix _before_ you do the tag / release"? Otherwise lgtm

Reply:
Updated now to the new wording!